src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -260,6 +261,70 @@ public static void BuildChain_WithCertificatePolicy_NoMatch()
         }
 
         [Fact]
+        [OuterLoop(/* Modifies user certificate store */)]
+        public static void BuildChain_WithRootCertInUserAndRootCertStores()
+        {
+            // Verifies that if a root cert is placed in both a root and 
+
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var microsoftDotComRoot = new X509Certificate2(TestData.MicrosoftDotComRootBytes))
+            using (var machineRootStore = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
+            using (var userRootStore = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+            {
+                try
+                {
+                    machineRootStore.Open(OpenFlags.ReadOnly);
+                    userRootStore.Open(OpenFlags.ReadWrite);
+
+                    // Check that the certificate is installed on the local machine
+                    var rootCerts = machineRootStore.Certificates.Find(X509FindType.FindByThumbprint, microsoftDotComRoot.Thumbprint, false);
+                    Assert.True(
+                        rootCerts.Count == 1, 
+                        string.Format("Expected to find exactly '1' root cert with thumbprint {0} in machine root certificate store", microsoftDotComRoot.Thumbprint));
+
+                    // Check that the certificate is NOT installed for the user
+                    var userCerts = userRootStore.Certificates.Find(X509FindType.FindByThumbprint, microsoftDotComRoot.Thumbprint, false);
+                    Assert.True(
+                        userCerts.Count == 0, 
+                        string.Format("Pre-install to user store - expected to find exactly '0' certs with thumbprint {0} in user root certificate store", microsoftDotComRoot.Thumbprint));
+
+                    userRootStore.Add(microsoftDotComRoot);
+
+                    // Check that the root certificate is installed for the user
+                    userCerts = userRootStore.Certificates.Find(X509FindType.FindByThumbprint, microsoftDotComRoot.Thumbprint, false);
+                    Assert.True(
+                        userCerts.Count == 1, 
+                        string.Format("Post-install to user store - expected to find exactly '1' certs with thumbprint {0} in user root certificate store", microsoftDotComRoot.Thumbprint));
+
+                    X509Chain chainValidator = new X509Chain();
```

Comment:
This is disposable, but please use ChainHolder (also Disposable) so it disposes the certs at the end of the test.